prov/efa/src/efa_av.c

```diff
@@ -459,75 +481,47 @@ void efa_conn_rdm_deinit(struct efa_av *av, struct efa_conn *conn)
  * 			Otherwise, return a negative libfabric error code
  */
 static
-int efa_av_update_connid_map(struct efa_av *av, struct efa_ep_addr *raw_addr,
-			      struct efa_conn *conn)
+int efa_av_update_reverse_av(struct efa_av *av, struct efa_ep_addr *raw_addr,
+				    struct efa_conn *conn)
 {
-	struct efa_connid_map *connid_map_entry = NULL;
-	struct efa_connid_map_key connid_map_key;
-
-	assert(av->ep_type == FI_EP_RDM);
-
-	memset(&connid_map_key, 0, sizeof(connid_map_key));
-	connid_map_key.ahn = conn->ah->ahn;
-	connid_map_key.qpn = raw_addr->qpn;
+	struct efa_cur_reverse_av *cur_entry = NULL;
+	struct efa_prv_reverse_av *prv_entry = NULL;
+	struct efa_cur_reverse_av_key cur_key;
+
+	memset(&cur_key, 0, sizeof(cur_key));
+	cur_key.ahn = conn->ah->ahn;
+	cur_key.qpn = raw_addr->qpn;
+	cur_entry = NULL;
+
+	HASH_FIND(hh, av->cur_reverse_av, &cur_key, sizeof(cur_key), cur_entry);
+	if (!cur_entry) {
+		cur_entry = malloc(sizeof(*cur_entry));
+		if (!cur_entry) {
+			FI_WARN(&rxr_prov, FI_LOG_AV, "Cannot allocate memory for cur_reverse_av entry");
+			return -FI_ENOMEM;
+		}
 
-	connid_map_entry = NULL;
-	HASH_FIND(hh, av->connid_map, &connid_map_key, sizeof(connid_map_key), connid_map_entry);
-	if (connid_map_entry) {
-		connid_map_entry->connid = raw_addr->qkey;
+		cur_entry->key.ahn = cur_key.ahn;
+		cur_entry->key.qpn = cur_key.qpn;
+		cur_entry->conn = conn;
+		HASH_ADD(hh, av->cur_reverse_av, key, sizeof(cur_key), cur_entry);
 		return 0;
```

Comment:
Is it expected that we return here for both rdm and dgram endpoint or just dgram endpoint?

Reply:
because we used a static connid for dgram endpoint, for dgram endpoint cur_entry should always be NULL